vm/vm.go

```diff
@@ -437,3 +532,66 @@ func (vm *Engine) Execute() error {
 
 	return vm.validateStateTransition()
 }
+
+// checkLockTime checks the absolute and relative lock time of the previous
+// asset. `blockTimestamp` is ignored for now.
+func checkLockTime(ctx context.Context, newAsset *asset.Asset,
+	witness *asset.Witness, blockHeight uint32, blockMeanTime time.Time,
+	chainLookup asset.ChainLookup) error {
+
+	// Check absolute lock time. This is easy as we can just compare the
+	// input asset's lock time to the current block height that we are aware
+	// of.
+	if newAsset.LockTime != 0 {
+		switch {
+		// If the lock time is a timestamp, we need to parse it as such
+		// and compare it to the current block height.
+		case newAsset.LockTime > txscript.LockTimeThreshold:
+			prevAssetTime := time.Unix(int64(newAsset.LockTime), 0)
+			if prevAssetTime.Before(blockMeanTime.UTC()) {
+				return newErrKind(ErrUnFinalizedAsset)
+			}
+
+		// Otherwise, we can just compare the lock time to the current
+		// block height.
+		case blockHeight < uint32(newAsset.LockTime):
+			return newErrKind(ErrUnFinalizedAsset)
+		}
+	}
+
+	// Now check any relative lock time. For this we need to look up the
+	// height of the block the input's anchor transaction was confirmed in.
+	if newAsset.RelativeLockTime != 0 {
```

Comment:
We assume `RelativeLockTime` is always expressed in blocks? The sequence field on btc anchorTx has a type flag for indicating blocks or seconds. (by the way, PR 3/3 also assumes blocks for `RelativeLockTime` but it also assumes blocks for `LockTime`!

Reply:
Hmm, good point. Not sure if we would want to also use a specific bit to indicate a time based sequence. Or if we also want to use the trick that everything above 500 million is a Unix timestamp... Added a TODO for now.

Reply:
Okay, added the same rules as in `btcd`. Re-using those functions wasn't really possible without restructuring everything, so I lifted some more logic instead.